pkg/endpoint/endpoint.go

```diff
@@ -2143,8 +2150,9 @@ func (e *Endpoint) RegenerateAfterCreation(ctx context.Context, syncBuild bool)
 		// proxy configuration, this code would effectively deadlock addition
 		// of endpoints.
 		e.Regenerate(&regeneration.ExternalRegenerationMetadata{
-			Reason:        "Initial build on endpoint creation",
-			ParentContext: ctx,
+			Reason:            "Initial build on endpoint creation",
+			ParentContext:     ctx,
+			RegenerationLevel: regeneration.RegenerateWithoutDatapath, // Needs datapath load??
```

Comment:
When an endpoint is created it needs to be regenerated at least from the template. Rewrite will take care of this.

```suggestion
			RegenerationLevel: regeneration.RegenerateWithDatapathRewrite,
```

Reply:
Refactored with this change.